File: 1]. Data Structures/01]. Array/cpp/01_twoSum.cpp
#include<bits/stdc++.h>

using namespace std;

// Your provided Two Sum function
vector<int> twoSum(vector<int>& nums, int target) {
    unordered_map<int, int> mp;  
    for (int i = 0; i < nums.size(); i++) {
        int complement = target - nums[i];
        if (mp.find(complement) != mp.end()) {
            return {mp[complement], i};
        }
        mp[nums[i]] = i; 
    }
    return {}; 
}

int main() {
    // Optimize standard I/O operations for competitive programming
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n, target;
    
    // Read the number of elements and the target sum
    if (cin >> n >> target) {
        vector<int> nums(n);
        
        // Read the array elements
        for (int i = 0; i < n; i++) {
            cin >> nums[i];
        }

        // Call your function
        vector<int> result = twoSum(nums, target);

        // Print the output
        if (!result.empty()) {
            cout << result[0] << " " << result[1] << "\n";
        } else {
            // Print -1 -1 if no valid pair is found
            cout << "-1 -1\n"; 
        }
    }

    return 0;
}

File: 1]. Data Structures/01]. Array/python/01_twoSum.py
import sys

def twoSum(nums,target):
    num_map = {};
    for i, num in enumerate(nums):
        complement = target - num
        if complement in num_map:
            return [num_map[complement], i]
        num_map[num] = i
    return [-1, -1]

def main():

    input_data = sys.stdin.read().split()

    if not input_data:
        print("-1 -1")
        return
    n =  int(input_data[0])
    target = int(input_data[1])

    nums = [int(x) for x in input_data[2:2+n]]

    result = twoSum(nums, target)

    if result != [-1, -1]:
        print(result[0], result[1])
    else:
        print("-1 -1")
    

if __name__ == "__main__":
    main()

    